backend/src/main/java/com/google/sps/GmailFactory.java

```diff
@@ -0,0 +1,69 @@
+package main.java.com.google.sps;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.extensions.java6.auth.oauth2.AuthorizationCodeInstalledApp;
+import com.google.api.client.extensions.jetty.auth.oauth2.LocalServerReceiver;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.GmailScopes;
+import com.google.common.collect.ImmutableList;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.util.List;
+
+public class GmailFactory {
+
+  /**
+   * Global instance of the scopes required by this quickstart. If modifying these scopes, delete
+   * your previously saved tokens/ folder.
+   */
+  private static final List<String> SCOPES = ImmutableList.of(GmailScopes.MAIL_GOOGLE_COM);
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String APPLICATION_NAME = "Ad-lib";
+  private static final String CREDENTIALS_FILE_PATH = "backend/credentials.json";
+  private static final String TOKENS_DIRECTORY_PATH = "tokens";
+  private final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+
+  public GmailFactory() throws GeneralSecurityException, IOException {}
```

Comment:
Question: Does Java implicit constructor so you don't have to declare an empty one like this? Also, will it throw any exceptions?

Reply:
Java does implicit constructor but due to the method calls within the class (get credentials) the constructor is used just for the sake of throwing errors which will occur via the HTTTP_Transport

Reply:
I would discourage throwing checked exceptions in the constructor for exactly this reason. The throws clause is contagious and now the servlets also have to define a default constructor that throws these exceptions. WDYT about creating the transport where it's needed instead of creating ti at construction time?

Reply:
Okay, that would make sense